atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/transaction/impl/SnapshotTransaction.java

```diff
@@ -1950,8 +1950,10 @@ private void commitWrites(TransactionService transactionService) {
                 if (validationNecessaryForInvolvedTablesOnCommit()) {
                     throwIfImmutableTsOrCommitLocksExpired(null);
                 }
-                return;
             }
+            long microsSinceCreation = TimeUnit.MILLISECONDS.toMicros(System.currentTimeMillis() - timeCreated);
+            getTimer("nonWriteCommitTotalTimeSinceTxCreation")
+                    .update(Duration.of(microsSinceCreation, ChronoUnit.MICROS));
```

Comment:
There is an existing transaction timer context we should use here and maintain higher  (sub millisecond) timer resolution. I'd also suggest a slightly different name

```suggestion
            getTimer("readOnlyTransactionDuration")
                    .update(transactionTimerContext.stop(), TimeUnit.NANOSECONDS);
```

Reply:
I agree that sub-milli second precision is preferred. Was keeping it consistent with the [already existing metric for txns that do writes](https://github.com/palantir/atlasdb/blob/629316c8b93bd3cfc7a27ab34c40fc28fecb997b/atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/transaction/impl/SnapshotTransaction.java#L2029), but I guess we can update that too!

Regarding the name: it was intentionally named that way, so people don't confuse it with the actual [ReadTransaction,](https://github.com/palantir/atlasdb/blob/629316c8b93bd3cfc7a27ab34c40fc28fecb997b/atlasdb-client/src/main/java/com/palantir/atlasdb/transaction/impl/ReadTransaction.java#L39) which are a different thing.